monai/data/nifti_saver.py

```diff
@@ -131,7 +132,7 @@ def save(self, data: Union[torch.Tensor, np.ndarray], meta_data: Optional[Dict]
         """
         filename = meta_data[Key.FILENAME_OR_OBJ] if meta_data else str(self._data_index)
         self._data_index += 1
-        original_affine = meta_data.get("original_affine", None) if meta_data else None
+        original_affine = meta_data.get("original_affine", None) if meta_data and self.resample else None
```

Comment:
Hi @wyli @ericspod ,

In the previous code, if `resample` is False, we only correct the orientation based on original affine and don't resample image size, does this PR change the behavior to even skip orientation? May I know why?

Thanks in advance.

Reply:
At this level of saver, I feel ignoring the original affine is a more common use case.

Reply:
So `resample` also includes `orientation` logic? I don't know the common practice in medical domain specific tasks, just want to double confirm with others @ericspod @rijobro .

Thanks.